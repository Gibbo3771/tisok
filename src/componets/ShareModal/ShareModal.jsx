import React from "react";
import ReactModal from "react-modal";
import PropTypes from "prop-types";

// All share stuff
import { FacebookShareButton, TwitterShareButton } from "react-share";
import { FacebookIcon, TwitterIcon } from "react-share";

export default class ShareModal extends React.Component {
  render() {
    const {
      isOpen,
      url,
      facebookQuote,
      facebookHashtag,
      twitterTitle,
      twitterHashtags
    } = this.props;
    return (
      <ReactModal id="modal" isOpen={this.props.isOpen}>
        <FacebookShareButton
          url={url}
          quote={facebookQuote}
          hashtag={facebookHashtag}
        >
          <FacebookIcon size={48} round={true} />
        </FacebookShareButton>
      </ReactModal>
    );
  }
}

ShareModal.propTypes = {
  url: PropTypes.string.isRequired,
  facebookQuote: PropTypes.string.isRequired,
  facebookHashtag: PropTypes.string.isRequired,
  twitterTitle: PropTypes.string.isRequired,
  twitterHashtags: PropTypes.array.isRequired
};
