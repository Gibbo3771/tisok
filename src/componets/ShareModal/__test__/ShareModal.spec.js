// @flow
import React from "react";
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
import ShareModal from "../ShareModal";
import { shallow, mount } from "enzyme";
import sinon from "sinon";
import ReactModal from "react-modal";

Enzyme.configure({ adapter: new Adapter() });

let wrapper = null;

const props = {
  url: "www.google.com",
  slip: {
    slip_id: 1,
    advice: "eat fruit"
  },
  isModalOpen: false,
  onRequestClose: sinon.spy()
};

const { url, slip, isModalOpen, onRequestClose } = props;

describe("ShareModal", () => {
  beforeEach(() => {
    mount(<div id="root" />);
    ReactModal.setAppElement(document.getElementById("root"));
  });
  it("renders correctly", () => {
    const wrapper = shallow(
      <ShareModal
        isOpen={isModalOpen}
        url={url}
        slip={slip}
        onRequestClose={onRequestClose}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("has text to share", () => {
    const wrapper = shallow(
      <ShareModal
        isOpen={isModalOpen}
        url={url}
        slip={slip}
        onRequestClose={onRequestClose}
      />
    );
    expect(wrapper.state().text).toEqual("Advice to live by...");
  });
  it("has hashtags", () => {
    const wrapper = shallow(
      <ShareModal
        isOpen={isModalOpen}
        url={url}
        slip={slip}
        onRequestClose={onRequestClose}
      />
    );
    expect(wrapper.state().hashtags).toBeDefined();
    expect(wrapper.state().hashtags.length).toEqual(2);
    expect(wrapper.state().hashtags[0]).toEqual("tisok");
    expect(wrapper.state().hashtags[1]).toEqual("lifeprotips");
  });
  xit("toggles open and closed", () => {
    const wrapper = shallow(
      <ShareModal
        isOpen={isModalOpen}
        url={url}
        slip={slip}
        onRequestClose={onRequestClose}
      />
    );
    expect(wrapper.instance().props.isOpen).toBeFalsy();
    wrapper.setProps({ isOpen: true });
    expect(wrapper.instance().props.isOpen).toBeTruthy();
  });
});
