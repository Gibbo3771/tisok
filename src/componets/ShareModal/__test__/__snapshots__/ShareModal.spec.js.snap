// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareModal renders correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="share-modal"
  closeTimeoutMS={0}
  id="modal"
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <CreatedButton
    className="share-facebook"
    hashtag="tisok"
    quote="Advice to live by..."
    url="www.google.com"
    windowHeight={400}
    windowWidth={550}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={48}
    />
  </CreatedButton>
  <CreatedButton
    className="share-twitter"
    hashtags={
      Array [
        "tisok",
        "lifeprotips",
      ]
    }
    title="Advice to live by..."
    url="www.google.com"
    windowHeight={400}
    windowWidth={550}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={48}
    />
  </CreatedButton>
  <ShareLink
    slip={
      Object {
        "advice": "eat fruit",
        "slip_id": 1,
      }
    }
  />
</Modal>
`;
